libtelemetry/src/main/java/com/mapbox/services/android/telemetry/MapboxTelemetry.java

```diff
@@ -184,7 +188,7 @@ private void sendEvents(List<Event> events, Callback httpCallback) {
   }
 
   private boolean startTelemetry() {
-    if (!isTelemetryEnabled) {
+    if (!isTelemetryEnabled && checkLocationPermission()) {
```

Comment:
Adding the check here would mean that https://github.com/mapbox/mapbox-events-android/blob/a2ec6948397250aa784ee9f4e478c64d646c5eb3/libtelemetry/src/main/java/com/mapbox/services/android/telemetry/MapboxTelemetry.java#L190-L192 won't be executed unless the user has the permissions granted and we don't want that.

The expected behavior is to have everything working except the location-related stuff.
We added the following `// TODO` comments to analyze if it would make sense to add the check within `TelemetryService` or not:
https://github.com/mapbox/mapbox-events-android/blob/a2ec6948397250aa784ee9f4e478c64d646c5eb3/libtelemetry/src/main/java/com/mapbox/services/android/telemetry/TelemetryService.java#L39
https://github.com/mapbox/mapbox-events-android/blob/a2ec6948397250aa784ee9f4e478c64d646c5eb3/libtelemetry/src/main/java/com/mapbox/services/android/telemetry/TelemetryService.java#L51
The idea of not creating the service if the permissions are not granted is great but we should keep the rest of functionalities working.

Now it's a good moment to extract that block of code into a well-named method.